scripts/spack/packages/mfem/package.py

```diff
@@ -164,6 +164,55 @@ class Mfem(Package, CudaPackage, ROCmPackage):
     variant('miniapps', default=False,
             description='Build and install miniapps')
 
+    # Custom variant not yet in upstream Spack; see
+    # https://github.com/spack/spack/pull/30210 for inspiration. Eventually,
+    # this variant should be replaced with a feature in upstream Spack that
+    # enables targeted use of ASan flags. The motivation for this variant is
+    # to inject ASan flags in this package, but not in its dependents;
+    # injecting ASan flags into specs will propagate them to MFEM's
+    # dependencts, which can result in undesirable behavior (e.g., building
+    # CMake with ASan flags).
+    variant('asan', default=False, description='Add Address Sanitizer flags')
+
+    # ASan is only supported by GCC and (some) LLVM-derived
```

Comment:
How about:
```python
# ASan is the abbreviated name for memory error checking capabilities in modern compilers such 
# as GCC and some LLVM-derived compilers. When compiled with ASan enabled, the generated 
# code can detect numerous memory errors such as use after free, heap overflows, stack overflows,
# use outside of scope, and memory leaks.
```

Reply:
Do you want that paragraph to replace just the first line, or the entire block comment?

Reply:
I would prefer a small comment. I personally find large comments to be distracting.

Reply:
Made comments briefer in [`88a20795`](https://github.com/LLNL/serac/pull/699/commits/88a207950b448cdb2e3feb81d359e2a4218186d6).